cmd/instance_pool_create.go

```diff
@@ -0,0 +1,183 @@
+		}
+
+		zone, err := getZoneByName(zoneName)
+		if err != nil {
+			return err
+		}
+
+		so, err := cmd.Flags().GetString("service-offering")
+		if err != nil {
+			return err
+		}
+
+		servOffering, err := getServiceOfferingByName(so)
+		if err != nil {
+			return err
+		}
+
+		templateFilterCmd, err := cmd.Flags().GetString("template-filter")
+		if err != nil {
+			return err
+		}
+		templateFilter, err := validateTemplateFilter(templateFilterCmd)
+		if err != nil {
+			return err
+		}
+
+		templateName, err := cmd.Flags().GetString("template")
+		if err != nil {
+			return err
+		}
+
+		template, err := getTemplateByName(zone.ID, templateName, templateFilter)
+		if err != nil {
+			return err
+		}
+
+		keypair, err := cmd.Flags().GetString("keypair")
+		if err != nil {
+			return err
+		}
+
+		if keypair == "" {
+			keypair = gCurrentAccount.DefaultSSHKey
+		}
+
+		size, err := cmd.Flags().GetInt("size")
+		if err != nil {
+			return err
+		}
+
+		sg, err := cmd.Flags().GetStringSlice("security-group")
+		if err != nil {
+			return err
+		}
+
+		securityGroups, err := getSecurityGroups(sg)
+		if err != nil {
+			return err
+		}
+
+		privnet, err := cmd.Flags().GetStringSlice("privnet")
+		if err != nil {
+			return err
+		}
+
+		privnets, err := getPrivnetList(privnet, zone.ID)
+		if err != nil {
+			return err
+		}
+
+		userDataPath, err := cmd.Flags().GetString("cloud-init")
+		if err != nil {
+			return err
+		}
+
+		userData := ""
+
+		if userDataPath != "" {
+			userData, err = getUserData(userDataPath)
+			if err != nil {
+				return err
+			}
+
+			if len(userData) >= maxUserDataLength {
+				return fmt.Errorf("user-data maximum allowed length is %d bytes", maxUserDataLength)
+			}
+		}
+
+		r := asyncTasks([]task{task{
+			egoscale.CreateInstancePool{
+				Name:              args[0],
+				Description:       description,
+				ZoneID:            zone.ID,
+				ServiceOfferingID: servOffering.ID,
+				TemplateID:        template.ID,
+				KeyPair:           keypair,
+				Size:              size,
+				SecurityGroupIDs:  securityGroups,
+				NetworkIDs:        privnets,
+				UserData:          userData,
+			},
+			fmt.Sprintf("Creating instance pool %q", args[0]),
+		}})
+		errs := filterErrors(r)
+		if len(errs) > 0 {
+			return errs[0]
+		}
+		pool := r[0].resp.(*egoscale.CreateInstancePoolResponse)
+
+		if !gQuiet {
+			return showInstancePool(pool.ID.String())
+		}
+
+		return nil
+
+	},
+}
+
+func init() {
+	// Required Flags
+
+	instancePoolCreateCmd.Flags().StringP("zone", "z", "", "Instance pool zone")
+	instancePoolCreateCmd.MarkFlagRequired("zone")
+	instancePoolCreateCmd.Flags().StringP("service-offering", "o", "", serviceOfferingHelp)
+	instancePoolCreateCmd.MarkFlagRequired("service-offering")
+	instancePoolCreateCmd.Flags().StringP("template", "t", "", "Instance pool template")
+	instancePoolCreateCmd.MarkFlagRequired("template")
+	instancePoolCreateCmd.Flags().IntP("size", "", 0, "Number of instance in the pool")
```

Comment:
Setting `2` as default value for Instance Pool size seems conservative enough and more relevant than an empty pool.

```suggestion
	instancePoolCreateCmd.Flags().IntP("size", "", 2, "Number of instances in the pool")
```